File: eCinema/eCinema.Services/Database/Entities/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCinema.Services.Database.Entities
{
    public class Review
    {
        public int Id { get; set; }
        
        [Required]
        [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5")]
        public int Rating { get; set; } // 1-5 stars
        
        [MaxLength(500)]
        public string? Comment { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? ModifiedAt { get; set; }
        
        public bool IsActive { get; set; } = true;

        public bool IsDeleted { get; set; } = false;
        
        public bool IsSpoiler { get; set; } = false;
        
        public bool IsEdited { get; set; } = false;
        
        [Required]
        public int UserId { get; set; }
        
        [Required]
        public int MovieId { get; set; }
        
        [ForeignKey(nameof(UserId))]
        public virtual User User { get; set; } = null!;
        
        [ForeignKey(nameof(MovieId))]
        public virtual Movie Movie { get; set; } = null!;
    }
} 

File: UI/ecinema_mobile/lib/models/review.dart
import 'package:json_annotation/json_annotation.dart';
import 'movie.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  int? id;
  int? userId;
  int? movieId;
  int? rating;
  String? comment;
  bool? isDeleted;
  bool? isSpoiler;
  DateTime? createdAt;
  DateTime? modifiedAt;
  bool? isEdited;
  String? userName;
  String? userImage; 

  Review({
    this.id,
    this.userId,
    this.movieId,
    this.rating,
    this.comment,
    this.isDeleted,
    this.isSpoiler,
    this.createdAt,
    this.modifiedAt,
    this.isEdited,
    this.userName,
    this.userImage,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);
  Map<String, dynamic> toJson() => _$ReviewToJson(this);
} 